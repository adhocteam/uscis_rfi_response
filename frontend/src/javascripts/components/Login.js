import React from "react";
import AuthService from "../services/Auth";
class Login extends React.Component {
  state = {
    redirectToReferrer: false,
    username: "",
    password: ""
  };
  login = e => {
    e.preventDefault();
    sessionStorage.setItem("token", "fhq412p89fqfp48h");
  };
  render() {
    if (AuthService.validToken()) {
      this.props.history.push("/private");
    }

    return (
      <div className="Login">
        <h2> Login </h2>
        <p> Please enter your email and password </p>
        <p>You must log in to view the page</p>
        <form onSubmit={this.login}>
          <label>
            Username
            <input
              type="text"
              value={this.state.username}
              onChange={e => this.setState({ username: e.target.value })}
            />
          </label>
          <br />
          <label>
            Password
            <input
              type="password"
              value={this.state.password}
              onChange={e => this.setState({ password: e.target.value })}
            />
          </label>
          <br />
          <input type="submit" value="Log In" />
        </form>
      </div>
    );
  }
}
export default Login;
