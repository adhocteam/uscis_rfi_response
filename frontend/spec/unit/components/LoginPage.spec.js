jest.mock("services/UscisApiService");

import React from "react";
import { shallow } from "enzyme";
import Login from "components/LoginPage.js";
import history from "services/history.js";

test("renders", () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
});

test("redirects on successful login", () => {
  const wrapper = shallow(<Login />);
  history.replace = jest.fn();
  wrapper.setProps({
    location: { search: "" },
  });
  wrapper.setState({
    email: "fake.admin@adhocteam.us",
    password: "thisisafakepassword",
  });
  wrapper.instance().login({
    preventDefault: jest.fn(),
  });
  return Promise.resolve().then(() => {
    expect(sessionStorage.length).toEqual(4);
    expect(history.replace).toHaveBeenCalled();
  });
});
