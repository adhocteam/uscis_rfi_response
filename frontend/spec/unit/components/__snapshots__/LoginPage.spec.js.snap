// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="login ds-l-container ds-u-padding-top--3 ds-u-sm-text-align--center ds-u-sm-text-align--left qa-uscis-upload-page"
>
  <h2>
    Login
  </h2>
  <p>
    Please enter your email and password
  </p>
  <p>
    You must log in to view the page
  </p>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        className="ds-c-field"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Password
      <input
        className="ds-c-field"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <br />
    <Button
      type="submit"
      variation="primary"
    >
      Submit
    </Button>
  </form>
</div>
`;
